unsigned int tomo::IoLoop::epoll_pass<true, true, false>(int)
--- PRE

+++ POST

@@ -21,183 +21,183 @@

 004a [ 2] test %eax,%eax
 004c [ 4] cmovns %eax,%r12d
 0050 [ 7] mov %r12d,0xc18(%r15)
-0057 [ 6] jle <+0x14f>
-005d [ 7] mov 0x190(%r15),%rax
-0064 [ 3] movslq %r12d,%rdx
-0067 [ 2] xor %ebp,%ebp
-0069 [ 7] lea <rip>(%rip),%r13 # <tomo_db0::IoLoop::r7_admit_fd<true, true>(int, tomo_db0::UrKind)::kErr+0x564>
-0070 [ 7] add %rdx,0x520(%rax)
-0077 [ 7] lea 0xac8(%r15),%rax
-007e [ 5] mov %rax,0x8(%rsp)
-0083 [ 5] lea 0x20(%rsp),%rax
-0088 [ 4] mov %rax,(%rsp)
-008c [ 4] nopl 0x0(%rax)
-0090 [ 4] mov 0x4(%rbx),%rdx
-0094 [ 3] mov %rdx,%rax
+0057 [ 2] jle <+0xcb>
+0059 [ 7] mov 0x190(%r15),%rax
+0060 [ 3] movslq %r12d,%rdx
+0063 [ 2] xor %ebp,%ebp
+0065 [ 7] lea <rip>(%rip),%r13 # <tomo_db0::(anonymous namespace)::kinds+0x77c>
+006c [ 7] add %rdx,0x520(%rax)
+0073 [ 7] lea 0xac8(%r15),%rax
+007a [ 5] mov %rax,0x8(%rsp)
+007f [ 5] lea 0x20(%rsp),%rax
+0084 [ 4] mov %rax,(%rsp)
+0088 [ 8] nopl 0x0(%rax,%rax,1)
+0090 [ 4] mov 0x4(%rbx),%rsi
+0094 [ 3] mov %rsi,%rax
 0097 [ 4] shr $0x38,%rax
 009b [ 2] cmp $0x11,%al
-009d [ 6] ja <+0x140>
-00a3 [ 5] movslq 0x0(%r13,%rax,4),%rax
-00a8 [ 3] add %r13,%rax
-00ab [ 3] notrack jmp *%rax
-00ae [ 2] mov (%rbx),%eax
-00b0 [ 5] test $0x2019,%eax
-00b5 [ 2] je <+0xbb>
-00b7 [ 4] movb $0x0,0x29(%rdx)
-00bb [ 2] test $0x1c,%al
-00bd [ 2] je <+0xe9>
-00bf [ 4] cmpb $0x0,0x30(%rdx)
-00c3 [ 5] mov %rdx,0x20(%rsp)
-00c8 [ 2] jne <+0xe9>
-00ca [ 5] lea 0x20(%rsp),%rsi
-00cf [ 7] lea 0x198(%r15),%rdi
-00d6 [ 4] movb $0x1,0x30(%rdx)
-00da [ 5] mov %rdx,0x10(%rsp)
-00df [ 5] call <std::deque<tomo::Client*, std::allocator<tomo::Client*> >::push_back(tomo::Client* const&)>
-00e4 [ 5] mov 0x10(%rsp),%rdx
-00e9 [ 6] mov 0x7bc(%rdx),%eax
-00ef [ 3] cmp $0xffffffff,%eax
-00f2 [ 2] je <+0x121>
-00f4 [ 7] mov 0x1a68(%r15),%rsi
-00fb [ 7] mov 0x1a70(%r15),%rcx
-0102 [ 3] sub %rsi,%rcx
-0105 [ 4] sar $0x3,%rcx
-0109 [ 3] cmp %rcx,%rax
-010c [ 2] jae <+0x121>
-010e [ 4] mov (%rsi,%rax,8),%rax
-0112 [ 3] test %rax,%rax
-0115 [ 2] je <+0x121>
-0117 [ 6] movw $0x0,0x3e(%rax)
-011d [ 4] movb $0x0,0x29(%rdx)
-0121 [ 3] mov %rdx,%rsi
-0124 [ 3] mov %r15,%rdi
-0127 [ 5] call <tomo::IoLoop::mark_active(tomo::Client*)>
-012c [ 4] nopl 0x0(%rax)
-0130 [ 3] inc %r14d
-0133 [11] data16 cs nopw 0x0(%rax,%rax,1)
-013e [ 2] xchg %ax,%ax
-0140 [ 2] inc %ebp
-0142 [ 4] add $0xc,%rbx
-0146 [ 3] cmp %r12d,%ebp
-0149 [ 6] jl <+0x90>
-014f [ 5] mov 0x28(%rsp),%rax
-0154 [ 9] sub %fs:0x28,%rax
-015d [ 6] jne <+0x428>
-0163 [ 4] add $0x38,%rsp
-0167 [ 3] mov %r14d,%eax
-016a [ 1] pop %rbx
-016b [ 1] pop %rbp
-016c [ 2] pop %r12
-016e [ 2] pop %r13
-0170 [ 2] pop %r14
-0172 [ 2] pop %r15
-0174 [ 1] ret
-0175 [11] data16 cs nopw 0x0(%rax,%rax,1)
-0180 [ 7] mov 0x190(%r15),%rax
-0187 [ 7] mov 0x188(%r15),%rdi
-018e [ 5] mov 0x8(%rsp),%rdx
-0193 [ 2] inc %ebp
-0195 [ 3] inc %r14d
-0198 [ 4] add $0xc,%rbx
-019c [ 2] mov (%rax),%esi
-019e [ 5] call <tomo::DatabaseMap::consume_wake(unsigned int, tomo::Ring&)>
-01a3 [ 3] cmp %r12d,%ebp
-01a6 [ 6] jl <+0x90>
-01ac [ 2] jmp <+0x14f>
-01ae [ 2] xchg %ax,%ax
-01b0 [ 7] mov 0x188(%r15),%rax
-01b7 [ 7] movzbl 0x115e0(%rax),%eax
-01be [ 2] cmp $0x3,%al
-01c0 [ 6] ja <+0x140>
-01c6 [ 7] mov 0xaa4(%r15),%eax
-01cd [ 4] mov %eax,0x10(%rsp)
-01d1 [ 2] test %eax,%eax
-01d3 [ 6] js <+0x140>
-01d9 [ 8] movl $0x0,0x1c(%rsp)
-01e1 [ 4] mov 0x10(%rsp),%edi
-01e5 [ 2] xor %edx,%edx
-01e7 [ 2] xor %esi,%esi
-01e9 [ 5] mov $0x80800,%ecx
-01ee [ 5] call <accept4@plt>
-01f3 [ 2] test %eax,%eax
-01f5 [ 6] jns <+0x410>
-01fb [ 5] call <__errno_location@plt>
-0200 [ 2] mov (%rax),%eax
-0202 [ 3] cmp $0x4,%eax
-0205 [ 6] jne <+0x393>
-020b [ 2] jmp <+0x1e1>
-020d [ 3] nopl (%rax)
-0210 [ 7] mov 0x188(%r15),%rax
-0217 [ 7] movzbl 0x115e0(%rax),%eax
-021e [ 2] cmp $0x3,%al
-0220 [ 6] ja <+0x140>
-0226 [ 7] mov 0xaa8(%r15),%eax
-022d [ 4] mov %eax,0x10(%rsp)
-0231 [ 2] test %eax,%eax
-0233 [ 6] js <+0x140>
-0239 [ 8] movl $0x0,0x1c(%rsp)
-0241 [ 4] mov 0x10(%rsp),%edi
-0245 [ 2] xor %edx,%edx
-0247 [ 2] xor %esi,%esi
-0249 [ 5] mov $0x80800,%ecx
-024e [ 5] call <accept4@plt>
-0253 [ 2] test %eax,%eax
-0255 [ 6] jns <+0x3d0>
-025b [ 5] call <__errno_location@plt>
-0260 [ 2] mov (%rax),%eax
-0262 [ 3] cmp $0x4,%eax
-0265 [ 6] jne <+0x393>
-026b [ 2] jmp <+0x241>
-026d [ 3] nopl (%rax)
-0270 [ 9] movq $0x0,0x20(%rsp)
-0279 [ 7] nopl 0x0(%rax)
-0280 [ 7] mov 0xbc8(%r15),%edi
-0287 [ 4] mov (%rsp),%rsi
-028b [ 5] mov $0x8,%edx
-0290 [ 5] call <read@plt>
-0295 [ 4] cmp $0x8,%rax
-0299 [ 2] je <+0x280>
-029b [ 7] mov 0x190(%r15),%rax
-02a2 [ 2] inc %ebp
-02a4 [ 3] inc %r14d
-02a7 [ 4] add $0xc,%rbx
-02ab [ 7] incq 0x440(%rax)
-02b2 [ 3] cmp %r12d,%ebp
-02b5 [ 6] jl <+0x90>
-02bb [ 5] jmp <+0x14f>
-02c0 [10] movabs $0xffffffffffff,%rax
-02ca [ 3] and %rax,%rdx
-02cd [ 6] je <+0x140>
-02d3 [ 6] mov 0x788(%rdx),%eax
-02d9 [ 7] mov 0x190(%r15),%rcx
-02e0 [ 2] cmp %eax,(%rcx)
-02e2 [ 6] jne <+0x140>
-02e8 [ 4] cmpb $0x0,0x33(%rdx)
-02ec [ 6] jne <+0x140>
-02f2 [ 6] mov 0x784(%rdx),%eax
-02f8 [ 3] cmp $0xfffffffe,%eax
-02fb [ 6] je <+0x140>
-0301 [ 7] mov 0x288(%r15),%rax
-0308 [ 7] mov 0x290(%r15),%rcx
-030f [ 3] cmp %rcx,%rax
-0312 [ 2] jne <+0x32d>
-0314 [ 5] jmp <+0xae>
-0319 [ 7] nopl 0x0(%rax)
-0320 [ 4] add $0x28,%rax
-0324 [ 3] cmp %rax,%rcx
-0327 [ 6] je <+0xae>
-032d [ 3] cmp (%rax),%rdx
-0330 [ 2] jne <+0x320>
-0332 [ 5] jmp <+0x140>
-0337 [ 9] nopw 0x0(%rax,%rax,1)
+009d [ 2] ja <+0xb0>
+009f [ 5] movslq 0x0(%r13,%rax,4),%rax
+00a4 [ 3] add %r13,%rax
+00a7 [ 3] notrack jmp *%rax
+00aa [ 6] nopw 0x0(%rax,%rax,1)
+00b0 [ 3] mov %r15,%rdi
+00b3 [ 5] call <tomo::IoLoop::fliplisten_epoll(unsigned long)>
+00b8 [ 3] add %eax,%r14d
+00bb [ 5] nopl 0x0(%rax,%rax,1)
+00c0 [ 2] inc %ebp
+00c2 [ 4] add $0xc,%rbx
+00c6 [ 3] cmp %r12d,%ebp
+00c9 [ 2] jl <+0x90>
+00cb [ 5] mov 0x28(%rsp),%rax
+00d0 [ 9] sub %fs:0x28,%rax
+00d9 [ 6] jne <+0x418>
+00df [ 4] add $0x38,%rsp
+00e3 [ 3] mov %r14d,%eax
+00e6 [ 1] pop %rbx
+00e7 [ 1] pop %rbp
+00e8 [ 2] pop %r12
+00ea [ 2] pop %r13
+00ec [ 2] pop %r14
+00ee [ 2] pop %r15
+00f0 [ 1] ret
+00f1 [11] data16 cs nopw 0x0(%rax,%rax,1)
+00fc [ 4] nopl 0x0(%rax)
+0100 [10] movabs $0xffffffffffff,%rax
+010a [ 3] and %rsi,%rax
+010d [ 2] je <+0xc0>
+010f [ 6] mov 0x788(%rax),%edx
+0115 [ 7] mov 0x190(%r15),%rcx
+011c [ 2] cmp %edx,(%rcx)
+011e [ 2] jne <+0xc0>
+0120 [ 4] cmpb $0x0,0x33(%rax)
+0124 [ 2] jne <+0xc0>
+0126 [ 6] mov 0x784(%rax),%edx
+012c [ 3] cmp $0xfffffffe,%edx
+012f [ 2] je <+0xc0>
+0131 [ 7] mov 0x288(%r15),%rdx
+0138 [ 7] mov 0x290(%r15),%rcx
+013f [ 3] cmp %rcx,%rdx
+0142 [ 2] je <+0x162>
+0144 [11] data16 cs nopw 0x0(%rax,%rax,1)
+014f [ 1] nop
+0150 [ 3] cmp (%rdx),%rax
+0153 [ 6] je <+0xc0>
+0159 [ 4] add $0x28,%rdx
+015d [ 3] cmp %rdx,%rcx
+0160 [ 2] jne <+0x150>
+0162 [ 2] mov (%rbx),%edx
+0164 [ 6] test $0x2019,%edx
+016a [ 2] je <+0x170>
+016c [ 4] movb $0x0,0x29(%rax)
+0170 [ 3] and $0x1c,%edx
+0173 [ 2] je <+0x19f>
+0175 [ 4] cmpb $0x0,0x30(%rax)
+0179 [ 5] mov %rax,0x20(%rsp)
+017e [ 2] jne <+0x19f>
+0180 [ 5] lea 0x20(%rsp),%rsi
+0185 [ 7] lea 0x198(%r15),%rdi
+018c [ 4] movb $0x1,0x30(%rax)
+0190 [ 5] mov %rax,0x10(%rsp)
+0195 [ 5] call <std::deque<tomo::Client*, std::allocator<tomo::Client*> >::push_back(tomo::Client* const&)>
+019a [ 5] mov 0x10(%rsp),%rax
+019f [ 6] mov 0x7bc(%rax),%edx
+01a5 [ 3] cmp $0xffffffff,%edx
+01a8 [ 2] je <+0x1d7>
+01aa [ 7] mov 0x1a68(%r15),%rsi
+01b1 [ 7] mov 0x1a70(%r15),%rcx
+01b8 [ 3] sub %rsi,%rcx
+01bb [ 4] sar $0x3,%rcx
+01bf [ 3] cmp %rcx,%rdx
+01c2 [ 2] jae <+0x1d7>
+01c4 [ 4] mov (%rsi,%rdx,8),%rdx
+01c8 [ 3] test %rdx,%rdx
+01cb [ 2] je <+0x1d7>
+01cd [ 6] movw $0x0,0x3e(%rdx)
+01d3 [ 4] movb $0x0,0x29(%rax)
+01d7 [ 3] mov %rax,%rsi
+01da [ 3] mov %r15,%rdi
+01dd [ 5] call <tomo::IoLoop::mark_active(tomo::Client*)>
+01e2 [11] data16 cs nopw 0x0(%rax,%rax,1)
+01ed [ 3] nopl (%rax)
+01f0 [ 3] inc %r14d
+01f3 [ 5] jmp <+0xc0>
+01f8 [ 8] nopl 0x0(%rax,%rax,1)
+0200 [ 7] mov 0x190(%r15),%rax
+0207 [ 7] mov 0x188(%r15),%rdi
+020e [ 5] mov 0x8(%rsp),%rdx
+0213 [ 3] inc %r14d
+0216 [ 2] mov (%rax),%esi
+0218 [ 5] call <tomo::DatabaseMap::consume_wake(unsigned int, tomo::Ring&)>
+021d [ 5] jmp <+0xc0>
+0222 [11] data16 cs nopw 0x0(%rax,%rax,1)
+022d [ 3] nopl (%rax)
+0230 [ 7] mov 0x188(%r15),%rax
+0237 [ 7] movzbl 0x115e0(%rax),%eax
+023e [ 2] cmp $0x3,%al
+0240 [ 6] ja <+0xc0>
+0246 [ 7] mov 0xaa4(%r15),%eax
+024d [ 4] mov %eax,0x10(%rsp)
+0251 [ 2] test %eax,%eax
+0253 [ 6] js <+0xc0>
+0259 [ 8] movl $0x0,0x1c(%rsp)
+0261 [ 4] mov 0x10(%rsp),%edi
+0265 [ 2] xor %edx,%edx
+0267 [ 2] xor %esi,%esi
+0269 [ 5] mov $0x80800,%ecx
+026e [ 5] call <accept4@plt>
+0273 [ 2] test %eax,%eax
+0275 [ 6] jns <+0x400>
+027b [ 5] call <__errno_location@plt>
+0280 [ 2] mov (%rax),%eax
+0282 [ 3] cmp $0x4,%eax
+0285 [ 6] jne <+0x393>
+028b [ 2] jmp <+0x261>
+028d [ 3] nopl (%rax)
+0290 [ 7] mov 0x188(%r15),%rax
+0297 [ 7] movzbl 0x115e0(%rax),%eax
+029e [ 2] cmp $0x3,%al
+02a0 [ 6] ja <+0xc0>
+02a6 [ 7] mov 0xaa8(%r15),%eax
+02ad [ 4] mov %eax,0x10(%rsp)
+02b1 [ 2] test %eax,%eax
+02b3 [ 6] js <+0xc0>
+02b9 [ 8] movl $0x0,0x1c(%rsp)
+02c1 [ 4] mov 0x10(%rsp),%edi
+02c5 [ 2] xor %edx,%edx
+02c7 [ 2] xor %esi,%esi
+02c9 [ 5] mov $0x80800,%ecx
+02ce [ 5] call <accept4@plt>
+02d3 [ 2] test %eax,%eax
+02d5 [ 6] jns <+0x3c0>
+02db [ 5] call <__errno_location@plt>
+02e0 [ 2] mov (%rax),%eax
+02e2 [ 3] cmp $0x4,%eax
+02e5 [ 6] jne <+0x393>
+02eb [ 2] jmp <+0x2c1>
+02ed [ 3] nopl (%rax)
+02f0 [ 9] movq $0x0,0x20(%rsp)
+02f9 [ 7] nopl 0x0(%rax)
+0300 [ 7] mov 0xbc8(%r15),%edi
+0307 [ 4] mov (%rsp),%rsi
+030b [ 5] mov $0x8,%edx
+0310 [ 5] call <read@plt>
+0315 [ 4] cmp $0x8,%rax
+0319 [ 2] je <+0x300>
+031b [ 7] mov 0x190(%r15),%rax
+0322 [ 3] inc %r14d
+0325 [ 7] incq 0x440(%rax)
+032c [ 5] jmp <+0xc0>
+0331 [11] data16 cs nopw 0x0(%rax,%rax,1)
+033c [ 4] nopl 0x0(%rax)
 0340 [ 7] mov 0x188(%r15),%rax
 0347 [ 7] movzbl 0x115e0(%rax),%eax
 034e [ 2] cmp $0x3,%al
-0350 [ 6] ja <+0x140>
+0350 [ 6] ja <+0xc0>
 0356 [ 7] mov 0xaa0(%r15),%eax
 035d [ 4] mov %eax,0x10(%rsp)
 0361 [ 2] test %eax,%eax
-0363 [ 6] js <+0x140>
+0363 [ 6] js <+0xc0>
 0369 [ 8] movl $0x0,0x1c(%rsp)
 0371 [ 4] mov 0x10(%rsp),%edi
 0375 [ 2] xor %edx,%edx
@@ -205,7 +205,7 @@

 0379 [ 5] mov $0x80800,%ecx
 037e [ 5] call <accept4@plt>
 0383 [ 2] test %eax,%eax
-0385 [ 2] jns <+0x3f0>
+0385 [ 2] jns <+0x3e0>
 0387 [ 5] call <__errno_location@plt>
 038c [ 2] mov (%rax),%eax
 038e [ 3] cmp $0x4,%eax
@@ -215,32 +215,28 @@

 0398 [ 7] mov 0x190(%r15),%rax
 039f [ 7] incq 0x458(%rax)
 03a6 [ 4] mov 0x1c(%rsp),%eax
-03aa [ 2] inc %ebp
-03ac [ 4] add $0xc,%rbx
-03b0 [ 3] add %eax,%r14d
-03b3 [ 3] cmp %r12d,%ebp
-03b6 [ 6] jl <+0x90>
-03bc [ 5] jmp <+0x14f>
-03c1 [11] data16 cs nopw 0x0(%rax,%rax,1)
-03cc [ 4] nopl 0x0(%rax)
-03d0 [ 5] mov $0x6,%edx
-03d5 [ 2] mov %eax,%esi
-03d7 [ 3] mov %r15,%rdi
-03da [ 4] incl 0x1c(%rsp)
-03de [ 5] call <void tomo::IoLoop::admit_fd<true, false>(int, tomo::UrKind)>
-03e3 [ 5] jmp <+0x241>
-03e8 [ 8] nopl 0x0(%rax,%rax,1)
-03f0 [ 5] mov $0x1,%edx
-03f5 [ 2] mov %eax,%esi
-03f7 [ 3] mov %r15,%rdi
-03fa [ 4] incl 0x1c(%rsp)
-03fe [ 5] call <void tomo::IoLoop::admit_fd<true, false>(int, tomo::UrKind)>
-0403 [ 5] jmp <+0x371>
-0408 [ 8] nopl 0x0(%rax,%rax,1)
-0410 [ 5] mov $0x9,%edx
-0415 [ 2] mov %eax,%esi
-0417 [ 3] mov %r15,%rdi
-041a [ 4] incl 0x1c(%rsp)
-041e [ 5] call <void tomo::IoLoop::admit_fd<true, false>(int, tomo::UrKind)>
-0423 [ 5] jmp <+0x1e1>
-0428 [ 5] call <__stack_chk_fail@plt>
+03aa [ 3] add %eax,%r14d
+03ad [ 5] jmp <+0xc0>
+03b2 [11] data16 cs nopw 0x0(%rax,%rax,1)
+03bd [ 3] nopl (%rax)
+03c0 [ 5] mov $0x6,%edx
+03c5 [ 2] mov %eax,%esi
+03c7 [ 3] mov %r15,%rdi
+03ca [ 4] incl 0x1c(%rsp)
+03ce [ 5] call <void tomo::IoLoop::admit_fd<true, false>(int, tomo::UrKind)>
+03d3 [ 5] jmp <+0x2c1>
+03d8 [ 8] nopl 0x0(%rax,%rax,1)
+03e0 [ 5] mov $0x1,%edx
+03e5 [ 2] mov %eax,%esi
+03e7 [ 3] mov %r15,%rdi
+03ea [ 4] incl 0x1c(%rsp)
+03ee [ 5] call <void tomo::IoLoop::admit_fd<true, false>(int, tomo::UrKind)>
+03f3 [ 5] jmp <+0x371>
+03f8 [ 8] nopl 0x0(%rax,%rax,1)
+0400 [ 5] mov $0x9,%edx
+0405 [ 2] mov %eax,%esi
+0407 [ 3] mov %r15,%rdi
+040a [ 4] incl 0x1c(%rsp)
+040e [ 5] call <void tomo::IoLoop::admit_fd<true, false>(int, tomo::UrKind)>
+0413 [ 5] jmp <+0x261>
+0418 [ 5] call <__stack_chk_fail@plt>
